src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], searchedKey: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const searchedKeys = new Set(items.map((value) => value[searchedKey]));
```

Comment:
There is no need to create a Set from array of keys. It creates unnecessary computational complexity.